cmd/server-mux.go

```diff
@@ -258,18 +324,19 @@ func (m *ServerMux) ListenAndServeTLS(certFile, keyFile string) (err error) {
 func (m *ServerMux) ListenAndServe() error {
 	go m.handleServiceSignals()
 
-	listener, err := net.Listen("tcp", m.Server.Addr)
+	listeners, err := initListeners(m.Server.Addr, &tls.Config{})
 	if err != nil {
 		return err
 	}
 
-	listenerMux := &ListenerMux{Listener: listener, config: &tls.Config{}}
-
 	m.mu.Lock()
-	m.listener = listenerMux
+	m.listeners = listeners
 	m.mu.Unlock()
 
-	err = m.Server.Serve(listenerMux)
+	for _, listener := range listeners[:len(listeners)-1] {
+		go m.Server.Serve(listener)
+	}
+	err = m.Server.Serve(listeners[len(listeners)-1])
```

Comment:
Same here. Why only check for error on the last listener alone?  It is worth commenting this.